### PR TITLE
Build updates

### PR DESCRIPTION
**Pull request by Xiche**
Various updates and fixes extracted from the previous incarnation of my fork and rebased.

### CODE REVIEW

**Comment by wtfbbqhax** [comment]
@Xiche `make dist` fails

**Comment by Xiche** [comment]
Fixed make dist in latest commit.

**Comment by wtfbbqhax** [comment]
Sent you more changes

**Review by wtfbbqhax — COMMENTED** [review]
This change is resulting in version "UNKNOWN" when generating distributions

**Comment by Xiche** [comment]
It relies on the repository being tagged.  My repository has v2.2.1 tagged, so it generates a version of 2.2.1-<#commits>-<abbreviated-SHA>.  The main repo is not yet tagged.

File: bootstrap [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#! /bin/sh
+set -x
+autoreconf -ivf --warnings=all
```
Comment (crc181) [inline bootstrap]: Needs $1 to build out of tree.

Comment (Xiche) [reply bootstrap]: I'm not sure what you're asking for; the autoreconf step cannot be done outside of the tree.

Comment (crc181) [reply bootstrap]: $ rm -rf temp
$ mkdir temp
$ cd temp
$ autoreconf -isvf ../abcip
autoreconf: Entering directory `../abcip'
autoreconf: configure.ac: not using Gettext
autoreconf: running: /opt/local/bin/aclocal --force -I m4
autoreconf: configure.ac: tracing
autoreconf: running: /opt/local/bin/glibtoolize --force

...

configure.ac:16: installing 'build-aux/compile'
configure.ac:12: installing 'build-aux/missing'
src/app/Makefile.am: installing 'build-aux/depcomp'
autoreconf: Leaving directory `../abcip'

$ ../abcip/configure 
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... ../abcip/build-aux/install-sh -c -d
...

Comment (Xiche) [reply bootstrap]: But why would you run autoreconf from a different directory?  It is no different than running it in the original directory.  The autoreconf script takes the last argument, pulls the directory name out of it, and then chdirs to it before running its subprograms.  I can add the $1, I'm just curious.

Comment (crc181) [reply bootstrap]: Because it is easy.  Why does autoreconf support that?

File: configure.ac [hunk 2]
```diff
@@ -2,8 +2,11 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_PREREQ([2.69])
-AC_INIT([abcip], [2.2.1], [crc181@gmail.com])
+AC_INIT([abcip],
```
Comment (crc181) [inline configure.ac]: So how do I assign a version with this change?  I want to set it to 2.3.0 after merge.

**Comment by crc181** [comment]
FYI - I did a rebase / merge which was probably the same as a standard merge in this case.

And configure and abcip --version are showing UNKNOWN version so I need to know how the git version magic works.

**Comment by Xiche** [comment]
Just add tags (annotated or not) to the repository.  Start out by posthumously tagging v2.2.1 (git tag v2.2.1 c360260fa26d9a584f4a86b5d04416ab53a0a70c) and then tag the next version when you're ready.  Push tags to GitHub with 'git push --tags origin'.

**Comment by crc181** [comment]
I was missing the 'v'.  At least it doesn't show up with --version.  I guess I can cope.  :)

If I had seen this first I would have made the latest 2.3.1 and the prior 2.3.0 ... Should be fixed now.

**Pull request merged**